src/Refactoring-UI/ReExtractToTemporaryDriver.class.st

```diff
@@ -0,0 +1,145 @@
+ReExtractToTemporaryDriver >> configureRefactoring [
+
+	refactoring := RBExtractToTemporaryRefactoring 
+							extract: sourceInterval
+						 	to: newVariableName
+							from: selector
+							in: class
+]
+
+{ #category : 'instance creation' }
+ReExtractToTemporaryDriver >> extract: anInterval to: aString from: aSelector in: aClass [
+
+	class := RBNamespace new classObjectFor: aClass.
+	selector := aSelector.
+	sourceInterval := anInterval.
+	newVariableName := aString
+]
+
+{ #category : 'initialization' }
+ReExtractToTemporaryDriver >> initialize [ 
+	
+	super initialize.
+	shouldEscape := false.
+]
+
+{ #category : 'accessing' }
+ReExtractToTemporaryDriver >> model: aModel [
+	model := aModel
+]
+
+{ #category : 'execution' }
+ReExtractToTemporaryDriver >> requestVariableName [
+
+	newVariableName := self defaultRequestDialog
+										title: 'Please provide a variable name';
+										label: 'The variable name should be defined.';
+										validateAnswer:[ :string :presenter |
+											self validateName: string onPresenter: presenter ];
+										openModal.
+	^ newVariableName 
+]
+
+{ #category : 'execution' }
+ReExtractToTemporaryDriver >> runRefactoring [
+
+	self configureRefactoring.
+	
+	refactoring prepareForExecution.
+	
+	newVariableName := self requestVariableName.
+	shouldEscape ifTrue: [ ^ self ].
+	
+	self applyChanges
+]
+
+{ #category : 'accessing' }
+ReExtractToTemporaryDriver >> scopes: aCollection [ 
+	scopes := aCollection
+]
+
+{ #category : 'initialization' }
+ReExtractToTemporaryDriver >> setErrorTextFromCondition: aCondition onPresenter: aPresenter [
+
+	stream := WriteStream on: String new.
+	aCondition violationMessageOn: stream.
+	aPresenter setErrorText: stream contents
+]
+
+{ #category : 'validation' }
+ReExtractToTemporaryDriver >> validateName: aName onPresenter: presenter [
+ 
+	| firstCondition secondCondition result errorMessages |
+   	firstCondition := refactoring preconditionValidInstVarName.
+	secondCondition := refactoring preconditionCheckVariableName.
+	errorMessages := OrderedCollection new.
+	
+	result := (firstCondition check and: [ secondCondition check ]).
+    
+   result 
+       ifFalse: [ 
+           firstCondition check ifFalse: [ 
+               self setErrorTextFromCondition: firstCondition onPresenter: presenter.
+               errorMessages add: (firstCondition violationMessageOn: stream).
+           ].
+           secondCondition check ifFalse: [ 
+               self setErrorTextFromCondition: secondCondition onPresenter: presenter.
+               errorMessages add: (secondCondition violationMessageOn: stream ).
+           ].
+           shouldEscape := true.
+       ]
+       ifTrue: [ presenter setErrorText: '' ].
+
+	errorMessages isEmpty
+		ifFalse: [ 
+           presenter setErrorText: (errorMessages join: ', ').
+       ]. 
+   
+	^ result.
+]
+
+{ #category : 'accessing' }
+ReExtractToTemporaryDriver >> variableNameEditorPresenterClass [
```

Comment:
Maybe this function's name should be `defaultRequestDialog`? Can you check how rename method did this mocking, might be helpful. If needed we can pair on this.

Reply:
I used rename method as an example, it has `methodNameEditorPresenterClass ` method.  Rename method uses a `requestDialogWith: ` and I just made a default one, I saw an example of it in the rename class refactoring. Now I see that my focus should be there, thank you @balsa-sarenac

Reply:
I thought so, thats why I asked if you need both mocks. If you need help, reach out. Try to debug rename method to see exactly how was mock set up, and then with that knowledge create mock specific to this use-case.